datafusion/datasource-parquet/src/row_filter.rs

```diff
@@ -520,111 +489,15 @@ mod test {
         let expr = col("int64_list").is_not_null();
         let expr = logical2physical(&expr, &table_schema);
 
-        let schema_adapter_factory = Arc::new(DefaultSchemaAdapterFactory);
         let table_schema = Arc::new(table_schema.clone());
 
-        let candidate = FilterCandidateBuilder::new(
-            expr,
-            table_schema.clone(),
-            table_schema,
-            schema_adapter_factory,
-        )
-        .build(metadata)
-        .expect("building candidate");
+        let candidate = FilterCandidateBuilder::new(expr, table_schema.clone())
+            .build(metadata)
+            .expect("building candidate");
 
         assert!(candidate.is_none());
     }
 
-    #[test]
-    fn test_filter_type_coercion() {
```

Comment:
Is there any way to keep this test (or something like it) that shows reading with a predicate of a different shcema is correctly coerced?

Reply:
I think so - I'll have to make it more e2e since it is no longer specific to the row filter. We have some other similar tests, I'll work this into there